File: app/src/main/java/com/example/registrationwithserver/RecordsActivity.java
package com.example.registrationwithserver;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class RecordsActivity extends AppCompatActivity {
    RecyclerView recordList;
    ArrayList<RegisteredData> registeredDataArrayList;
    RecordAdapter recordAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_records);
        registeredDataArrayList = getIntent().getParcelableArrayListExtra("data");
        recordAdapter = new RecordAdapter(this, registeredDataArrayList);
        setViews();
        configureRecyclerView();
    }

    private void configureRecyclerView() {
        RecyclerView.LayoutManager layoutManager = new GridLayoutManager(this, 1);
        recordList.setLayoutManager(layoutManager);
        recordList.setHasFixedSize(true);
        recordList.setAdapter(recordAdapter);
        recordAdapter.notifyDataSetChanged();
    }

    private void setViews() {
        recordList = findViewById(R.id.record_list);
    }
}

File: app/src/main/java/com/example/registrationwithserver/RecordsActivity.kt
package com.example.registrationwithserver

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.registrationwithserver.databinding.ActivityRecordsBinding
import java.util.*

class RecordsActivity : AppCompatActivity() {
    var registeredDataArrayList: ArrayList<RegisteredData>? = null
    var recordAdapter: RecordAdapter? = null
    private lateinit var binding: ActivityRecordsBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityRecordsBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)
        registeredDataArrayList = intent.getParcelableArrayListExtra("data")
        recordAdapter = RecordAdapter(this, registeredDataArrayList)
        configureRecyclerView()
    }

    private fun configureRecyclerView() {
        val layoutManager: RecyclerView.LayoutManager = GridLayoutManager(this, 1)
        binding.recordList.layoutManager = layoutManager
        binding.recordList.setHasFixedSize(true)
        binding.recordList.adapter = recordAdapter
        recordAdapter!!.notifyDataSetChanged()
    }

}